datastream-kafka/src/test/java/com/linkedin/datastream/kafka/TestKafkaProducerWrapper.java

```diff
@@ -141,8 +153,17 @@ void verifyFlush(int numExpected) {
       verify(_mockProducer, times(numExpected)).flush();
     }
 
-    void verifyClose(int numExpected) {
+    void verifyClose(int numExpected) throws NoSuchMethodException {
+      // Producer close is invoked in a separate thread. Must wait for the thread to get scheduled and call close
+      Method method = Producer.class.getMethod("close", long.class, TimeUnit.class);
+      PollUtils.poll(() -> {
+        Collection<Invocation> invocations = mockingDetails(_mockProducer).getInvocations();
+        long count = invocations.stream().filter(invocation -> invocation.getMethod().equals(method)).count();
+        return count == numExpected;
+      }, 1000, 10000);
       verify(_mockProducer, times(numExpected)).close(anyLong(), any(TimeUnit.class));
+      Assert.assertEquals(_numShutdownProducerCalls, numExpected);
+      _numShutdownProducerCalls = 0;
```

Comment:
Unlike the other two `verify*()` methods, this one resets its call count (`_numShutdownProducerCalls`). This is a little confusing because the call counts passed to the other methods in `testFlushInterrupt()` are all cumulative.

Reply:
As discussed offline, leaving this as is since mockProducers can be created and removed during the test